crates/bevy_asset/src/assets.rs

```diff
@@ -50,22 +50,16 @@ impl AssetIndex {
 pub(crate) struct AssetIndexAllocator {
     /// A monotonically increasing index.
     next_index: AtomicU32,
-    recycled_queue_sender: Sender<AssetIndex>,
+    recycled_queue: ConcurrentQueue<AssetIndex>,
     /// This receives every recycled [`AssetIndex`]. It serves as a buffer/queue to store indices ready for reuse.
-    recycled_queue_receiver: Receiver<AssetIndex>,
-    recycled_sender: Sender<AssetIndex>,
-    recycled_receiver: Receiver<AssetIndex>,
+    recycled: ConcurrentQueue<AssetIndex>,
```

Comment:
I think this refactor is 100% worth doing, regardless of whether we end up removing the other channels. Right here the queues have clear a clear owner so it only makes sense to remove the indirection incurred by the channel